app/javascript/controllers/turbo_controller.js

```diff
@@ -0,0 +1,21 @@
+import { Controller } from "@hotwired/stimulus"
+
+/*
+ * TurboController is used to handle the Turbo events and to
+ * add some custom behavior to the Turbo navigation.
+ */
+export default class extends Controller {
```

Comment:
I think this is generally useful behavior that shouldn't be limited to specific pages. In other projects I have a "root controller" that is attached to the body element where I can put behavior like this. It's not 100% in the Stimulus approach (where you pass specific elements in), but the other option would be to have to remember to do this for every form that has turbo enabled (which eventually should be all of them).

Reply:
I bound this onto the `body` element in the layout. Are you saying you suggest having an overarching one that doesn't care which layout it is? I could insert this as javascript that just loads on every page but doesn't do anything unless `turbo:submit-end` occurs.

Reply:
Yeah exactly - on connect it should just listen for `turbo:submit-end` and do its thing.

Reply:
Gotcha! I can refactor this to put this logic inside the `connect` so we don't have to specify the `data-action`